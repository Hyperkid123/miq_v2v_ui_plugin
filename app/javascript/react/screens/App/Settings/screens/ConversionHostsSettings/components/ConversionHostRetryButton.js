import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'patternfly-react';

const ConversionHostRetryButton = ({
  task,
  setConversionHostTaskToRetryAction,
  showConversionHostRetryModalAction,
  ...props
}) => (
  <Button
    id={`retry_${task.id}`}
    onClick={e => {
      e.stopPropagation();
      setConversionHostTaskToRetryAction(task);
      showConversionHostRetryModalAction();
    }}
    {...props}
  >
    {__('Retry')}
  </Button>
);

ConversionHostRetryButton.propTypes = {
  task: PropTypes.object,
  setConversionHostTaskToRetryAction: PropTypes.func,
  showConversionHostRetryModalAction: PropTypes.func
};

export default ConversionHostRetryButton;
