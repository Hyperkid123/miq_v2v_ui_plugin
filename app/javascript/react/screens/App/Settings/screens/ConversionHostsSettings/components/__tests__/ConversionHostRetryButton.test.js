import React from 'react';
import { shallow } from 'enzyme';
import ConversionHostRetryButton from '../ConversionHostRetryButton';

describe('conversion host retry button', () => {
  const getBaseProps = () => ({
    task: { id: '12345', name: 'mock-task' },
    setConversionHostTaskToRetryAction: jest.fn(),
    showConversionHostRetryModalAction: jest.fn()
  });

  it('renders correctly', () => {
    const component = shallow(<ConversionHostRetryButton {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
  });

  it('renders with a unique id for each task', () => {
    const c1 = shallow(<ConversionHostRetryButton {...getBaseProps()} />);
    const c2 = shallow(<ConversionHostRetryButton {...getBaseProps()} task={{ id: '67890', name: 'mock-task' }} />);
    expect(c1.find('Button').props().id).not.toEqual(c2.find('Button').props().id);
  });

  it('passes extra props to the Button', () => {
    const component = shallow(<ConversionHostRetryButton {...getBaseProps()} extraProp="extraValue" />);
    expect(component.find('Button').props().extraProp).toBe('extraValue');
  });

  it('calls the correct actions on click', () => {
    const props = getBaseProps();
    const component = shallow(<ConversionHostRetryButton {...props} />);
    const stopPropagation = jest.fn();
    component.find('Button').simulate('click', { stopPropagation });
    expect(stopPropagation).toHaveBeenCalledTimes(1);
    expect(props.setConversionHostTaskToRetryAction).toHaveBeenCalledTimes(1);
    expect(props.setConversionHostTaskToRetryAction).toHaveBeenCalledWith(props.task);
    expect(props.showConversionHostRetryModalAction).toHaveBeenCalledTimes(1);
  });
});
