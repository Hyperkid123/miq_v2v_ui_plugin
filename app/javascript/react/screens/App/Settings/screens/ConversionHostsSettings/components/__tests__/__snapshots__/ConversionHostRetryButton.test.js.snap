// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion host retry button renders correctly 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  id="retry_12345"
  onClick={[Function]}
>
  Retry
</Button>
`;
